File: Cmdlets/GetCultureInfoCommand.vb
Imports System.Globalization
Imports System.Management.Automation

<Cmdlet("Get", "CultureInfo")>
Public Class GetCultureInfoCommand
	Inherits Cmdlet

	<Parameter()>
	Public Property Name() As String

	<Parameter()>
	Public Property All() As SwitchParameter

	<Parameter()>
	Public Property CultureTypes() As CultureTypes = CultureTypes.AllCultures

	Protected Overrides Sub BeginProcessing()

		If Me.All.IsPresent Then

			Me.WriteObject(CultureInfo.GetCultures(Me.CultureTypes), True)
		ElseIf Me.Name IsNot Nothing Then

			Me.WriteObject(New CultureInfo(Me.Name))
		Else

			Dim CurrentCultures As New System.Collections.Generic.Dictionary(Of String, CultureInfo)

			CurrentCultures.Add("CurrentCulture", CultureInfo.CurrentCulture)
			CurrentCultures.Add("CurrentUICulture", CultureInfo.CurrentUICulture)

			Me.WriteObject(CurrentCultures, True)
		End If
	End Sub
End Class

File: Cmdlets.Core/Commands/GetCultureInfoCommand.cs
using System.Globalization;
using System.Management.Automation;

namespace TIKSN.Cmdlets.Core.Commands
{
    [Cmdlet("Get", "CultureInfo")]
    public class GetCultureInfoCommand : Cmdlet
    {
        public GetCultureInfoCommand()
        {
            CultureTypes = CultureTypes.AllCultures;
        }

        [Parameter()]
        public SwitchParameter All { get; set; }

        [Parameter()]
        public CultureTypes CultureTypes { get; set; }

        [Parameter()]
        public string Name { get; set; }

        protected override void BeginProcessing()
        {
            if (this.All.IsPresent)
            {
                this.WriteObject(CultureInfo.GetCultures(this.CultureTypes), true);
            }
            else if (this.Name != null)
            {
                this.WriteObject(new CultureInfo(this.Name));
            }
            else
            {
                System.Collections.Generic.Dictionary<string, CultureInfo> CurrentCultures = new System.Collections.Generic.Dictionary<string, CultureInfo>();

                CurrentCultures.Add("CurrentCulture", CultureInfo.CurrentCulture);
                CurrentCultures.Add("CurrentUICulture", CultureInfo.CurrentUICulture);

                this.WriteObject(CurrentCultures, true);
            }
        }
    }
}